### PR TITLE
Add public `createTrustManagers` factory

### PR DESCRIPTION
**Pull request by markelliot**

### CODE REVIEW

**Comment by markelliot** [comment]
@uschi2000 I think we need this to play nicely with new okhttp

File: ssl-config/src/main/java/com/palantir/remoting/ssl/SslSocketFactories.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public static SSLContext createSslContext(SslConfiguration config) {
         }
     }
 
-    private static TrustManagerFactory createTrustManagerFactory(
+    public static TrustManagerFactory createTrustManagerFactory(
```
Comment (uschi2000) [inline ssl-config/src/main/java/com/palantir/remoting/ssl/SslSocketFactories.java]: the trustStorePath parameter is non-trivial (can be file or directory, depending on the type), so this will need docs, unfortunately.

File: ssl-config/src/main/java/com/palantir/remoting/ssl/SslSocketFactories.java [hunk 2]
```diff
@@ -63,7 +61,17 @@ public static SSLContext createSslContext(SslConfiguration config) {
         }
     }
 
-    public static TrustManagerFactory createTrustManagerFactory(
+    /**
+     * Create a {@link TrustManager[]} initialized from the provided configuration.
+     *
+     * @param config an {@link SslConfiguration} describing at least the trust store configuration
+     * @return an {@link TrustManager[]} according to the input configuration
```
Comment (uschi2000) [inline ssl-config/src/main/java/com/palantir/remoting/ssl/SslSocketFactories.java:68]: ok, so I guess I'll have to add docs to SslConfiguration now ;)

**Comment by uschi2000** [comment]
👍

File: ssl-config/src/main/java/com/palantir/remoting/ssl/SslSocketFactories.java (hunk 1 above)
Comment (markelliot) [inline ssl-config/src/main/java/com/palantir/remoting/ssl/SslSocketFactories.java]: Refactored to make this match the other methods in the class in style and docs.

File: ssl-config/src/main/java/com/palantir/remoting/ssl/SslSocketFactories.java (hunk 2 above)
Comment (markelliot) [inline ssl-config/src/main/java/com/palantir/remoting/ssl/SslSocketFactories.java:68]: yep, but we're already doing this elsewhere, so we can leave that for another PR
